src/main/java/ArrayValueCalculator.java

```diff
@@ -0,0 +1,34 @@
+public class ArrayValueCalculator {
+    public int doCalc(String[][] array) throws ArraySizeException {
+        int sumOfElements = 0;
+        int lengthInput = array.length;
+        if (lengthInput != 4) {
+            throw new ArraySizeException("Invalid array length: " + lengthInput);
+        } else {
+            for (String[] row : array) {
+                if (row.length != 4) {
+                    throw new ArraySizeException("Invalid array row length: " + row.length);
+                }
+            }
+        }
+        int rowCounter = 0;
+        for (String[] row : array) {
+            int columnCounter = 0;
+            for (String value : row) {
+                try {
+                    sumOfElements += Integer.parseInt(value);
+                } catch (NumberFormatException e) {
+                    try {
```

Comment:
We usually don't have such constructions (`catch->continue`).

You created exception and on the next line hide it. In similar cases in code better to have like:
```
} catch (NumberFormatException e) {
      sout("Invalid value:" + " [" + rowCounter + "," + columnCounter + "]\n" + e);
}
```

Reply:
In any case, that is different from the requirements. You sum all numbers, if some are not numbers - you ignore them. In task - if any not number throw an exception, no sum